lib/pdk/module/templatedir.rb

```diff
@@ -82,7 +92,7 @@ def metadata
           'pdk-version' => PDK::Util::Version.version_string,
         }
 
-        result['template-url'] = @repo ? @repo : @path
+        result['template-url'] = @cloned_from
```

Comment:
When a user specifies a template-url and a template-ref branch, the intention is for the template to set the template-ref to `heads/<branch>` right? When I tested this, this is what I got in metadata.json.
```
  "pdk-version": "1.10.0.pre (remotes/hunner/add_template_ref-0-gf4f35df3c7)",
  "template-url": "https://github.com/bmjen/pdk-templates#pdk-1277",
  "template-ref": "heads/master-0-g9ebdc53"
```

Reply:
`template-ref` is set to the `git describe` output but is only used for UX purposes when informing the user about the template update
```
PDK (INFO): Updating <module> using the template at <template url> from <old SHA or tag> to <new SHA or tag>
```
The `template-ref` value is read and cached at the start of the update process to provide the `<old SHA or tag>` value.

Reply:
`template-ref` is not authoritative and just a description of the HEAD ref of the template as generated by `git describe --all --long`. When the description is based off a branch as above, we only care about the SHA at the end and not the branch name which is not guaranteed to be the branch you expect (as above where git figured out that `master` and `pdk-1277` branches share the same HEAD and so chose `heads/master` to base the ref description from).

Reply:
So, from a UX perspective, the output I showed is confusing. I've provided a template-ref of a branch name, and it appeared as if the `new module` command correctly cloned and rendered the module based on that branch. I'd also expect the `git describe --all --long` to show the branch name, so that's weird. Nonetheless, we should try to make this outward facing value make sense, if we only care about the 8char SHA at the end then we should parse that and construct a new string with the branch name or tag.

Reply:
@rodjek So this seems to still show inconsistency with the latest update. Still seeing this:
```
  "pdk-version": "1.10.0.pre (remotes/hunner/add_template_ref-0-gb6a9d585c8)",
  "template-url": "https://github.com/puppetlabs/pdk-templates#pin-puppet",
  "template-ref": "heads/master-0-ge07e4fd"
```

@scotje  @rodjek What's our stance on this? I'm not a fan of the strings that imply "branch" being inconsistent between the `template-url` and `template-ref` where one says `pin-puppet` and the other says `heads/master`.

Reply:
I've changed the logic so that we cache the template-ref before we reset the head to the desired ref, which means we now get a more consistent looking `template-ref` value.

```
semirhage :0: pdk/foo (git:add_template_ref → hunner {2} U:1 ?:1!)$ ../bin/pdk update --template-ref pin-puppet
pdk (INFO): Updating tsharpe-foo using the default template, from master@059d3b5 to pin-puppet@e07e4fd

[snip]

semirhage :0: pdk/foo (git:add_template_ref → hunner {2} U:1 ?:1!)$ tail -n 4 metadata.json
  "pdk-version": "1.10.0.pre (heads/add_template_ref-0-g35bc88e)",
  "template-url": "https://github.com/puppetlabs/pdk-templates#pin-puppet",
  "template-ref": "remotes/origin/pin-puppet-0-ge07e4fd"
}
```